Add masking training (#9140)
qgis/QGIS-Documentation: docs/training_manual/basic_map/symbology.rst

.. figure:: img/svg_symbol_result.png
   :align: center

:abbr:`★★☆ (Moderate level)` Follow Along: Masking
-------------------------------------------------------------------------------

Masking feature allows you to mask some symbol layer using an other symbol layer or label.

In the following exercise, we want to configure renderering so polygon labels will mask the black part
of lines layer.

#. Load the :file:`selective_masking.qgs` from dataset in directory :file:`exercise_data/masking`.

#. Open the :guilabel:`Layer Properties` window for the ``polys``
   layer.

>    Under the |labels| :guilabel:`Labels` tab, you will see that |labels| :guilabel:`Single Labels` have

Author: Done

   already been defined.

#. From the :guilabel:`Labels` tab, select the |mask| :guilabel:`Mask` (and not :guilabel:`Masks` in general category list)
   

> #. Enable label mask by clicking on the checkbox labeled :guilabel:`Enable mask`

Author: Ok, I'll try to remember

   .. note:: A message tells you that the now defined label mask shape can be later selected as a mask source to
	     mask other symbol layers.

#. Click :guilabel:`OK`
#. Open the :guilabel:`Layer Properties` window for the ``lines``
   layer.

#. Go to the |mask| :guilabel:`Masks` tab

#. Click on the black line part of lines checkbox from masked symbol layers

   .. note::  A message warns you that you have to select both sources and symbol layers to be able to save

> 	      masking configuration

Author: Yes, the warning message is weird (way better in french!), I'm gonna propose to rewrite it.

I changed the sentence this way 

```
A message warns you that you have to select both masking and masked symbol layers to be able to save
   masking configuration. Again, let's do that.
```

> 	      masking configuration

Author: see https://github.com/qgis/QGIS/pull/58135

#. Click on ``polys`` :guilabel:`Label mask` checkbox

   The message disappear, you can now safely save your configuration.

#. Click :guilabel:`OK`

> At the right zoom level you can see that labels are masking the black part of lines, remaining only the

Author: It's absolutely not related to zoom level, my point was to said that at some zoom level, labels mask line depending on where they are positioned but it's too ambiguous/weird. I removed this part and leave only

> You can see now that labels are masking the black part of lines, leaving only the
yellow part.

yellow part.
   
.. figure:: img/label_masking.png
   :align: center

> :abbr:`★★☆ (Moderate level)` Try Yourself:

Author: Agreed. Done

-------------------------------------------------------------------------------

We want now to mask the black part of lines using black only planes that represent the ``points`` layer.
For the sake of visibility, you can switch off ``polys`` visibility.

> .. admonition:: Solution

Reviewer: ```suggestion
.. admonition:: Answer
```
This is what we used to use, although "solution" sounds a bit better to me.

   :class: dropdown

   #. Open the :guilabel:`Layer Properties` window for the ``points``
      layer.

>       Under the |symbology| :guilabel:`Symbology` tab, you will see that a |categorized| :guilabel:`Categorized` renderer have
>       been defined with 3 different symbol depending on the ``Class`` field value.

Reviewer: ```suggestion
      Under the |symbology| :guilabel:`Symbology` tab, you will see that a |categorized| :guilabel:`Categorized` renderer has
      been defined with 3 different symbols depending on the ``Class`` field value.
```

>    #. Double click on ``B52`` plane icon, within the :guilabel:`Symbol` column

Reviewer: It could be nice to mention earlier that b52 and jet are the items that will be concerned by the masking adjustments

      .. figure:: img/plane_symbology.png
	 :align: center

   #. Click the |symbologyAdd| :sup:`Add symbol layer` button and select the ``Mask`` :guilabel:`Symbol layer type`

>       A new hierarchy ``Mask`` :sup:`symbol layer` > ``Marker`` :sup:`symbol` > ``Simple Marker`` :sup:`symbol layer` appear below the root symbol ``Marker``.

Reviewer: I don't know if the previous step needs a screenshot but I'm pretty sure this one deserves one. And it will probably avoids/helps skip the description in the last paragraph (I'm curious about the HTML rendering of this paragraph... and note that :sup: is meant for tooltips-like entries)

Author: Done 1a684fb390

I kept the paragraph as I wanted to explicitly tell what is a symbol and what is a symbol layer

      
      Now, let's define a mask that is identical to our plane marker but a little bit bigger.

   #. Select newly created ``Simple Marker`` symbol layer and change its :guilabel:`Symbol layer type` to ``SVG marker``
   

>    #. Set the SVG file path to be the same as the already existing SVG marker, and the size to be bigger (24 millimeters

Author: > Is it easier to find the appropriate svg file? 

from my point of view yes, it's just a copy/paste, but I can change for something like

```
Select the same SVG as the already existing SVG marker
```

Do you thing it's better ?

> Any directive?

What do you mean?

>    #. Set the SVG file path to be the same as the already existing SVG marker, and the size to be bigger (24 millimeters

Reviewer: I meant a way to find the right svg. But I'm fine with your feedback

      is a pretty good choice for this use case).

      Mask color doesn't matter here, only the shape is used in masking. Symbol layer order is also irrelevant in that
      situation.

      You should get something like that:
   

>       .. figure:: img/plane_symbology.png

Reviewer: ```suggestion
      .. figure:: img/point_masking.png
```

	 :align: center

   #. Repeat the same operations on ``Jet`` symbol.
	   
   #. Click :guilabel:`OK`
   #. Open the :guilabel:`Layer Properties` window for the ``lines``
      layer.

   #. Go to the |mask| :guilabel:`Masks` tab

      the black line part of lines checkbox from masked symbol layers is already checked from our previous
      configuration.

>    #. Uncheck the :guilabel:`Label mask` ``polys`` checkbox and check both ``B52`` and ``Jet`` :guilabel:`Mask symbol layer`

Reviewer: A screenshot?

Author: Done
